diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_trace.h b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_trace.h
index 869604768..0bf9e3312 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_trace.h
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_trace.h
@@ -23,6 +23,7 @@
  *
  */
 
+#if !defined(NOTRACE)
 #undef TRACE_SYSTEM
 #define TRACE_SYSTEM amdgpu_dm
 
@@ -603,3 +604,4 @@ TRACE_EVENT(amdgpu_dm_dce_clocks_state,
 #define TRACE_INCLUDE_PATH .
 #define TRACE_INCLUDE_FILE amdgpu_dm_trace
 #include <trace/define_trace.h>
+#endif  /* !defined(NOTRACE) */
